5638 Introduced Entry date filter for RECAP search

albertisfu opened the pull request:
This PR solves #5638

It adds a new filter for the Entry date filed to RECAP search, which works for both the search frontend and the API.

I added the filter for both versions, just like the case filter. Behind the same waffle flag, it displays either the calendar or the relative dates version. For regular users, it shows only the calendar filter for the new field.

I’ve refactored the HTML template to avoid duplicated code so that the same include can be reused for different date filters. The JavaScript has also been refactored to support multiple date filter widgets dynamically.

### Users who don’t have the relative dates flag enabled:

**Home RECAP Search.** 
![Screenshot 2025-06-06 at 11 05 40 a m](https://github.com/user-attachments/assets/1a91bd6e-be0e-4358-8f83-164445475a6a)

**Sidebar.**
![Screenshot 2025-06-06 at 11 06 56 a m](https://github.com/user-attachments/assets/624158b4-ae58-4a22-b910-9694bef63f2d)

### Users with the relative dates waffle flag enabled:

**Home RECAP Search.**
![Screenshot 2025-06-06 at 11 07 45 a m](https://github.com/user-attachments/assets/92b02922-16c0-441c-bf87-191584d8eeee)

**Sidebar.**
![Screenshot 2025-06-06 at 11 09 24 a m](https://github.com/user-attachments/assets/bf7273f6-a1f5-4d65-9284-a6c389ca0510)


**Ensure that percolator pagination returns all alerts consistently.**
I found a flaky test related to RECAP Search Alerts while working on this. The test occasionally missed an alert triggered by the percolator. The issue occurs because, during testing, alerts are created almost simultaneously, which causes some alerts to have the same sorting key for Elasticsearch pagination. This results in inconsistent pagination that can lead to a missed alert. While this issue would be rare in production, it's not impossible, particularly if users create alerts using the API. In this case, it's better to use the alert ID, which is unique, to sort them instead of `date_created`. I’ve applied this fix in this PR.

mlissner commented:
This looks great. A couple thoughts:

1. In #5638 I renamed the fields a bit, so they're now "Docket Date" and "Entry Date". I'm hopeful that'll make things a bit more clear. 
2. I toggled the waffle to be live for everybody. Sorry I didn't let you know. I think we can remove the waffle altogether, if you're ready to. 
3. Something that's been bothering me is that if you select from the drop down, it's not consistent about the relative values. I assume the idea was to teach people that whatever format works, but in practice it has been confusing me. Can we make the following changes:

 - `-1m ago` -> `1m ago`
 - `-3m ago` -> `3m ago`
 - etc. In other words, remove the negatives?

Thank you!

albertisfu commented:
@mlissner Sure I'll apply the changes.

Regarding this one:
> Something that's been bothering me is that if you select from the drop down, it's not consistent about the relative values. I assume the idea was to teach people that whatever format works, but in practice it has been confusing me. Can we make the following changes:
> -1m ago -> 1m ago
> -3m ago -> 3m ago
> etc. In other words, remove the negatives?

We should only remove the negative values from the Relative filter options but continue supporting the negative syntax, right?

mlissner commented:
> We should only remove the negative values from the Relative filter options but continue supporting the negative syntax, right?

Exactly, yeah.

albertisfu commented:
I've applied the requested changes:


`Docket Date` and `Entry Date`
![Screenshot 2025-06-06 at 5 37 34 p m](https://github.com/user-attachments/assets/8b1597b7-738f-43ca-babd-d8d9022dd46a)

![Screenshot 2025-06-06 at 5 37 50 p m](https://github.com/user-attachments/assets/04f518d7-4514-4747-8912-f6814d5a8fac)


- Removed the waffle logic completely.
- Changed the relative filter default options to:
```
<li data-value="1d ago">Past Day</li>
<li data-value="7d ago">Past Week</li>
<li data-value="14d ago">Past 2 Weeks</li>
<li data-value="1m ago">Past Month</li>
<li data-value="3m ago">Past 3 Months</li>
<li data-value="6m ago">Past 6 Months</li>
<li data-value="1y ago">Past Year</li> 
```

mlissner commented:
Thanks Alberto!

mlissner reviewed (approved):
Seems fine to me, but I only looked quickly.

ERosendo reviewed (approved):
LGTM :shipit:
